### PR TITLE
feat: add styling for integration with bootstrap forms

### PR DESCRIPTION
**Pull request by inikolova**
Related to https://github.com/telerik/kendo-theme-default/issues/450

### CODE REVIEW

File: scss/common/_forms.scss [hunk 1]
```diff
@@ -225,4 +225,89 @@
...
+    .k-widget.form-control .k-input {
+        height: $bootstrap-integration-input-height;
+    }
+
+    .k-maskedtextbox.form-control .k-textbox,
+    .k-textbox.form-control {
+        height: calc(#{$bootstrap-integration-input-height} + 2px);
+    }
+
+    /* included in more recent theme versions */
```
Comment (gyoshev) [inline scss/common/_forms.scss]: The styles below this comment should be considered for the date/time pickers in general, and are not related to the bootstrap integration. The pickers should support variable width.

File: scss/common/_forms.scss [hunk 2]
```diff
@@ -225,4 +225,89 @@
...
+        }
+    }
+
+    .k-widget.form-control .k-input,
+    .k-maskedtextbox.form-control .k-textbox,
+    .k-textbox.form-control {
+        padding: ($bootstrap-integration-spacer / 2) ($bootstrap-integration-spacer * .75);
+    }
+
+    $bootstrap-integration-input-height: $bootstrap-integration-spacer * 2.25 !default;
```
Comment (gyoshev) [inline scss/common/_forms.scss]: It appears that the fractions are based on the values available in the Bootstrap framework (`2.25`, for example, seems oddly specific). How is this going to be used in `kendo-theme-bootstrap`?

File: scss/common/_forms.scss (hunk 1 above)
Comment (inikolova) [reply scss/common/_forms.scss]: Ok, I will remove these from here :)

File: scss/common/_forms.scss (hunk 2 above)
Comment (inikolova) [reply scss/common/_forms.scss]: I tested the integration-with-boostrap-forms styling with both - Default and Bootstrap themes, and everything is rendered correctly. Please keep in mind that in order to get everything aligned, all Kendo UI for Angular components should be 38px high (this is the Bootstrap form control elements height). @gyoshev , @joneff what do you think? Am I missing sth?

**Comment by joneff** [comment]
Looks fine to me

**Review by joneff — APPROVED** [review]

File: scss/common/_forms.scss (hunk 2 above)
Comment (gyoshev) [reply scss/common/_forms.scss]: Bootstrap uses the variables `$input-padding-y`, `$input-padding-x`, and `$input-line-height` to size form inputs. With this in mind, you can define the spacing in this file like this:

    $twbs-font-size-base: 1rem !default;
    $twbs-input-padding-x: .5rem !default;
    $twbs-input-padding-y: .75rem !default;
    $twbs-input-line-height: 1.25 !default;

(values are copied from Bootstrap to work in kendo-theme-default without introducing a dependency to Bootstrap).
With these declarations, you can write the following lines:

    padding: ($bootstrap-integration-spacer / 2) ($bootstrap-integration-spacer * .75);
    $bootstrap-integration-input-height: $bootstrap-integration-spacer * 2.25;

as

    padding: $twbs-input-padding-y $twbs-input-padding-x;
    $bootstrap-integration-input-height: $twbs-input-padding-y * 2 + $twbs-font-size-base * $twbs-input-line-height;

While a bit more complex, the above mimics the Bootstrap variables, and reacts to changes in them. If customers load the bootstrap SCSS and modify it (like in kendo-theme-bootstrap), the values can be defined as:

    $twbs-font-size-base: $font-size-base !default;
    $twbs-input-padding-x: $input-padding-x !default;
    $twbs-input-padding-y: $input-padding-y !default;
    $twbs-input-line-height: $input-line-height !default;

Thus inheriting the variable values from Bootstrap.

Comment (inikolova) [reply scss/common/_forms.scss]: OK, I will make these changes :)

**Comment by gyoshev** [comment]
Looks good. I'll merge this in `develop` tomorrow.

**Pull request closed without merging**